src/actions/actions.py

```diff
@@ -28,6 +29,123 @@
+            action_args['physical-network'],
+            action_args['network-type'].lower(),
+            action_args['cidr'],
+            action_args.get('segmentation-id'),
+            action_args.get('destination-cidr'),
+            action_args.get('nexthop'),
+        )
+    except exceptions.DuplicateResource as ex:
+        msg = (
+            f"The '{ex.resource_type}' to be created by this action was "
+            "already duplicated. This will have to be cleaned up manually."
+        )
+        hookenv.log(f'{msg} Exception: {ex}')
+        return hookenv.action_fail(
+            f'{msg} Check the unit logs for more details.')
+    except exceptions.InvalidResource as ex:
+        msg = (
+            f"The '{ex.resource_type}' to be created by this action already "
+            "exists with different parameters than given to the action. This "
+            "will have to be cleaned up manually."
+        )
+        hookenv.log(f'{msg} Exception: {ex}')
+        return hookenv.action_fail(
+            f'{msg} Check the unit logs for more details.')
+    except exceptions.APIException as ex:
+        hookenv.log(
+            "Encountered exception while creating the Trove management "
+            f"network. Exception: {ex}")
+        return hookenv.action_fail(
+            'Neutron API may not be available yet, rerun action when it is. '
+            'Check the unit logs for more details.')
+
+
+def _validate_create_mgmt_net_action_args(action_args):
+    net_type = action_args['network-type'].lower()
+    if net_type not in ['flat', 'vlan']:
+        return hookenv.action_fail(
+            "Only flat or vlan network types are supported.")
+
+    segmentation_id = action_args['segmentation-id']
+    if net_type == 'flat' and segmentation_id != 0:
+        return hookenv.action_fail(
+            "Segmentation ID given for flat network.")
+
+    if net_type == 'vlan' and (segmentation_id < 1 or segmentation_id >= 4095):
+        return hookenv.action_fail(
+            f"Invalid segmentation ID given: {segmentation_id}")
+
+    if not _is_cidr(action_args['cidr']):
+        return hookenv.action_fail(
+            "'cidr' argument is invalid. Must be a proper CIDR.")
+
+    dest_cidr = action_args.get('destination-cidr')
+    nexthop = action_args.get('nexthop')
+    # destination-cidr and nexthop are optional, but if they're given, both of
+    # them need to be given.
+    if any([dest_cidr, nexthop]) and not all([dest_cidr, nexthop]):
+        return hookenv.action_fail(
+            "Either both 'destination-cidr' and 'nexthop' arguments must be "
+            "given, or neither.")
+
+    if dest_cidr and not _is_cidr(dest_cidr):
+        return hookenv.action_fail(
+            "'destination-cidr' argument is invalid. Must be a proper CIDR.")
+
+    if nexthop and not _is_ip(nexthop):
+        return hookenv.action_fail(
+            "'nexthop' argument is invalid. Must be an IP.")
+
+
+def _is_cidr(cidr):
```

Comment:
This should be moved to `utils.py`, so we can use it in another file as well

Reply:
Done